StripeFinancialConnections/StripeFinancialConnections/Source/API Bindings/FinancialConnectionsAPIClient.swift

```diff
@@ -8,8 +8,54 @@
 import Foundation
 @_spi(STP) import StripeCore
 
-protocol FinancialConnectionsAPIClient {
+class FinancialConnectionsAPIClient {
+    let backingAPIClient: STPAPIClient
 
+    var isLinkWithStripe: Bool = false
+    var consumerPublishableKey: String?
+    var consumerSession: ConsumerSessionData?
+
+    static let shared = FinancialConnectionsAPIClient(apiClient: .shared)
+
+    init(apiClient: STPAPIClient) {
+        self.backingAPIClient = apiClient
+    }
+
+    func consumerPublishableKeyProvider(canUseConsumerKey: Bool) -> String? {
+        guard canUseConsumerKey, isLinkWithStripe, consumerSession?.isVerified == true else {
+            return nil
+        }
+        return consumerPublishableKey
+    }
+
+    func get<T: Decodable>(
+        resource: String,
+        parameters: [String: Any],
+        canUseConsumerPublishableKey: Bool
+    ) -> Promise<T> {
+        let consumerPubKey = consumerPublishableKeyProvider(canUseConsumerKey: canUseConsumerPublishableKey)
+        return backingAPIClient.get(
+            resource: resource,
+            parameters: parameters,
+            ephemeralKeySecret: consumerPubKey
+        )
+    }
+
+    func post<T: Decodable>(
+        resource: String,
+        parameters: [String: Any],
+        canUseConsumerPublishableKey: Bool
+    ) -> Promise<T> {
+        let consumerPubKey = consumerPublishableKeyProvider(canUseConsumerKey: canUseConsumerPublishableKey)
```

Comment:
calling out different pattern of naming: 

`consumerPubKey` doesn't spell out the "publishable" while `canUseConsumerPublishableKey` does (I prefer the spelled out, personally [`conusmerPubKey` -> `consumerPublishableKey`])